test_runner/src/main.rs

```diff
@@ -1,3 +1,155 @@
+use core::time;
+use std::thread;
+
+use althea_kernel_interface::{KernelInterfaceError, KI};
+
 fn main() {
-    println!("Hello, world!");
+    /*
+    These are currently connected as such:
+    A---------B
+     \       /|
+      \     / |
+       \   /  |
+        \ /   |
+         X    |
+        / \   |
+       /   \  |
+      /     \ |
+     /       \|
+    D---------C
+    */
+    let testa = ("nA".to_string(), 0);
+    let testb = ("nB".to_string(), 1);
+    let testc = ("nC".to_string(), 2);
+    let testd = ("nD".to_string(), 3);
+
+    let ten_mins = time::Duration::from_secs(600);
+
+    let namespaces = NamespaceInfo {
+        names: vec![
+            testa.clone().0,
+            testb.clone().0,
+            testc.clone().0,
+            testd.clone().0,
+        ],
+        linked: vec![
+            // arbitrary connections 
+            (testa.clone(), testb.clone()),
+            (testb.clone(), testc.clone()),
+            (testa, testc.clone()),
+            (testc, testd.clone()),
+            (testb, testd),
+        ],
+    };
+    let res = setup_ns(namespaces);
+    println!("Namespaces setup: {:?}", res);
+    // this sleep is for debugging so that the container can be accessed to poke around in
+    thread::sleep(ten_mins);
+}
+
+fn setup_ns(spaces: NamespaceInfo) -> Result<(), KernelInterfaceError> {
+    // arbitrary number for the IP assignment
+    let mut counter = 6;
+    // clear namespaces
+    KI.run_command("ip", &["-all", "netns", "delete", "||", "true"])?;
+    // add namespaces
+    for name in spaces.names {
+        let res = KI.run_command("ip", &["netns", "add", &name]);
+        println!("{:?}", res);
+        // ip netns exec nB ip link set dev lo up
+        let res = KI.run_command(
+            "ip",
+            &[
+                "netns", "exec", &name, "ip", "link", "set", "dev", "lo", "up",
+            ],
+        );
+        println!("{:?}", res);
+    }
+    for link in spaces.linked {
+        let veth_ab = format!("veth-{}-{}", link.0 .0, link.1 .0);
+        let veth_ba = format!("veth-{}-{}", link.1 .0, link.0 .0);
+        let ip_ab = format!("192.168.{}.{}/24", link.0 .1, counter);
+        let ip_ba = format!("192.168.{}.{}/24", link.1 .1, counter);
+        let subnet_a = format!("192.168.{}.0/24", link.0 .1);
+        let subnet_b = format!("192.168.{}.0/24", link.1 .1);
```

Comment:
embedding what ip's to assign into the original struct is a potential future improvement, for now this is good.